client/X11/xf_window.c

```diff
@@ -185,30 +185,27 @@ void xf_SetWindowFullscreen(xfContext* xfc, xfWindow* window, BOOL fullscreen)
 		 */
 		startX = startX + xfc->instance->settings->MonitorLocalShiftX;
 		startY = startY + xfc->instance->settings->MonitorLocalShiftY;
-	}
-
-	xf_ResizeDesktopWindow(xfc, window, width, height);
 
-	/* Set the fullscreen state */
-	xf_SendClientEvent(xfc, window->handle, xfc->_NET_WM_STATE, 4,
-				fullscreen ? _NET_WM_STATE_ADD : _NET_WM_STATE_REMOVE,
-				xfc->_NET_WM_STATE_FULLSCREEN, 0, 0);
-
-	/* Only send monitor bounds if they are valid */
-	if ((xfc->fullscreenMonitors.top >= 0) &&
-			(xfc->fullscreenMonitors.bottom >= 0) &&
-			(xfc->fullscreenMonitors.left >= 0) &&
-			(xfc->fullscreenMonitors.right >= 0))
-	{
-		xf_SendClientEvent(xfc, window->handle, xfc->_NET_WM_FULLSCREEN_MONITORS, 5,
+		/* Set monitor bounds */
+		if (settings->MonitorCount > 1)
```

Comment:
I do not believe this is the correct thing to do. If the user specifies a single monitor on the command-line when they have multiple monitors then that request should be honored, as it specifies exactly which monitor to fullscreen the window on. I think the "settings->MonitorCount variable currently only keeps track of the number of monitors to display on and not the total monitors available. Does this case make sense to you?